lib/util/signal.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 Huawei Cloud Computing Technologies Co., Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "sync"
+
+type Signal struct {
+	mu     sync.RWMutex
```

Comment:
```suggestion
	mu     sync.Mutex
```

the read operations are all very very simple, the overhead of a RWMutex over a Mutex is higher than the operation itsself, that's using a Mutex will be better.